docs/builtins-reference.md

```diff
@@ -0,0 +1,176 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+## Table of Contents
+
+  * [Introduction](#introduction)
+  * [DML-bodied builtin functions](#dml-bodied-builtin-functions)
+    * [`lm`-function](#lm-function)
+    * [`lmDS`-function](#lmds-function)
+    * [`lmCG`-function](#lmcg-function)
+    * [`lmpredict`-function](#lmpredict-function)
+    
+# Introduction
+
+Declarative Machine Learning has builtin functions which enable access to both low- and high-level functions
+to support all kinds of use cases.
+
+Builtins are either implemented on a compiler level or as DML scripts that are loaded at compile time.
+
+# DML-bodied builtin functions
+
+**DML-bodied builtin functions** are written as DML-Scripts and executed as such when called.
+
+## `lm`-function
+
+The `lm`-function solves linear regression using either the **direct solve method** or the **conjugate gradient algorithm**
+depending on the input size of the matrices (See [`lmDS`-function](#lmds-function) and 
+[`lmCG`-function](#lmcg-function) respectively).
+
+### Usage
+```r
+lm(X, y, icpt = 0, reg = 1e-7, tol = 1e-7, maxi = 0, verbose = TRUE)
+```
+
+### Arguments
+| Name    | Type           | Default  | Description |
+| :------ | :------------- | -------- | :---------- |
+| X       | Matrix[Double] | required | Matrix of feature vectors. |
+| y       | Matrix[Double] | required | 1-column matrix of response values. |
+| icpt    | Integer        | `0`      | Intercept presence, shifting and rescaling the columns of X ([Details](#icpt-argument))|
+| reg     | Double         | `1e-7`   | Regularization constant (lambda) for L2-regularization. set to nonzero for highly dependant/sparse/numerous features|
+| tol     | Double         | `1e-7`   | Tolerance (epsilon); conjugate gradient procedure terminates early if L2 norm of the beta-residual is less than tolerance * its initial norm|
+| maxi    | Integer        | `0`      | Maximum number of conjugate gradient iterations. 0 = no maximum |
+| verbose | Boolean        | `TRUE`   | If `TRUE` print messages are activated |
+
+Note that if number of *features* is small enough (`rows of X/y < 2000`), the [`lmDS`-function'](#lmds-function)
+is called internally and parameters `tol` and `maxi` are ignored.
+
+### Returns
+| Type           | Description |
+| :------------- | :---------- |
+| Matrix[Double] | 1-column matrix of weights. |
+
+##### icpt-Argument
+
+The *icpt-argument* can be set to 3 modes:
+ 
+  * 0 = no intercept, no shifting, no rescaling
+  * 1 = add intercept, but neither shift nor rescale X
+  * 2 = add intercept, shift & rescale X columns to mean = 0, variance = 1
+
+### Example
+```r
+X = rand (rows = 50, cols = 10)
+y = rand (rows = 50, cols = 1)
```

Comment:
Please make it more meaningful by using `y = X %*% rand(rows=ncol(X), 1)` (for all examples)